File: Python/monotonic_array.py
def monotonic_array(array):

  if(len(array)<=1):
    return True
  left=array[0]
  right=array[len(array)-1]
  if(left>right):
    for i in range(len(array)-1):
      if(array[i+1]>array[i]):
        return False
  elif (left==right):
    for i in range(len(array)-1):
      if(array[i+1]!=array[i]):
        return False
  else:
    for i in range(len(array)-1):
      if(array[i+1]<array[i]):
        return False
  return True

print(monotonic_array([1,8,4]))

File: JavaScript/monotonic_array.js
function monotonicArray(array){
    if(array<=1) return true
    let left=array[0]
    let right=array[array.length-1]
    
    if(left>right){
      for(let i=0; i<array.length-1;i++){
        if(array[i]<array[i+1]) return false
      }
    }
    else if(left===right){
      for (let i=0; i<array.length-1; i++){
        if(array[i] !==array[i+1]) return false
      }
    }
    else{
      for (let i=0; i<array.length-1;i++){
        if(array[i]>array[i+1]) return false
      }
    }
    return true
  }
  
  console.log(monotonicArray([4,3,2,5]))